akka-cluster-sharding/src/main/resources/reference.conf

```diff
@@ -245,6 +245,19 @@ akka.cluster.sharding {
   # Mostly for the Akka test suite, if off the invalid transition is logged as a warning instead of throwing and
   # crashing the shard.
   fail-on-invalid-entity-state-transition = off
+
+  # Healthcheck that can be used with Akka management health checks: https://doc.akka.io/docs/akka-management/current/healthchecks.html
+  healthcheck {
+    # sharding names to check have registered with the coordinator for the health check to pass
+    # once initial registration has taken place the health check always returns true to prevent the coordinator
+    # moving making the health check of all nodes fail
+    # by default no sharding instances are monitored
+    names = []
```

Comment:
can't we use `ClusterSharding(system).shardTypeNames` to automatically include all instead of manual configuration?

Reply:
we could still have this config, and use the automatic when `names` is empty, otherwise specific names can be given if not all are critical

Reply:
On problem with the automatic names is that there is a chance that the first health check request arrives before the application has initialized sharding, so if we support that we have to wait until Up and recommend that sharding is initialized immediately at startup and not after MemberUp.

Maybe better to use the explicit configuration anyway?

Reply:
I understand the concern, but this is the kind of thing where people make silly mistakes. 

It would be better if we promoted the healthcheck to be added programatically instead of loading based on classpath and configs. Then users could add the names to objects / static fields and reference them. 

Alternatively, when init'ing the shard, we could check if the name is added to this settings and if not, we could emit a warning saying that that node can receive traffic while the region is not yet registered... but then we are tying it to healthcheck. People may use cluster without having a healthcheck at all. So, may not a good idea. 

I don't know, what else can we do to make it less error prone?

Reply:
The membership check seems almost mandatory. This one i think is optional. We'd need to change how health checks work to make it programatic

Reply:
I think it's good thinking by Renato to have one less config that could be misconfigured but it would make this feature more difficult for us to implement.
* new API additions (both Typed and classic)
* the race condition of sharding initalization by the app and the health check calls that may arrive earlier

Therefore I suggest that we start with this config based implementation and if we see that this is important we can add programatic registration later.

Reply:
👍